File: .claude/hooks/backend-lint-format.py
#!/usr/bin/env python
"""PostToolUse hook: gofmt + go vet on edited Go files."""
import json
import subprocess
import sys
from pathlib import Path

data = json.load(sys.stdin)
file_path = Path(data.get("tool_input", {}).get("file_path", ""))

if file_path.suffix != ".go":
    sys.exit(0)

backend_dir = Path(__file__).parent.parent.parent / "backend"

subprocess.run(["gofmt", "-w", str(file_path)], check=False)

result = subprocess.run(
    ["go", "vet", "./..."],
    cwd=backend_dir,
    capture_output=True,
    text=True,
)
if result.returncode != 0:
    print(result.stderr, file=sys.stderr)
    sys.exit(1)

File: .claude/hooks/backend-lint-format.js
#!/usr/bin/env node
/**
 * PostToolUse hook: gofmt + go vet on edited Go files.
 *
 * Exit codes:
 *   0 = allow
 *   1 = go vet failed (shown to Claude)
 */

'use strict';

const path = require('path');
const { spawnSync } = require('child_process');

const backendDir = path.join(__dirname, '..', '..', 'backend');

let data = '';
process.stdin.setEncoding('utf8');
process.stdin.on('data', chunk => { data += chunk; });
process.stdin.on('end', () => {
  let input = {};
  try { input = JSON.parse(data); } catch {}

  const filePath = (input && input.tool_input && input.tool_input.file_path) || '';
  if (!filePath.endsWith('.go')) process.exit(0);

  spawnSync('gofmt', ['-w', filePath], { stdio: 'ignore' });

  const result = spawnSync('go', ['vet', './...'], {
    cwd: backendDir,
    encoding: 'utf8',
  });

  if (result.status !== 0) {
    process.stderr.write(result.stderr || result.stdout || '');
    process.exit(1);
  }

  process.exit(0);
});
